src/components/course/data/hooks.jsx

```diff
@@ -137,23 +81,22 @@ export function useAllCourseData({
     };
     fetchData();
   }, [
+    courseService,
     activeCatalogs,
     canEnrollWithEnterpriseOffers,
     couponCodes,
-    courseKey,
-    courseRunKey,
-    enterpriseConfig,
     enterpriseOffers,
     subscriptionLicense,
   ]);
 
   return {
-    courseData: camelCaseObject(courseData),
-    courseRecommendations: camelCaseObject(courseRecommendations),
```

Comment:
Doing the `camelCaseObject` here creates a new object/list reference on every parent component's re-render. If using the `courseData` output from this hook in a `useEffect`'s dependency array, it'd cause it to constantly fire.

Moved the `camelCaseObject` call to where we set the data instead.